Source/LinqToDB/Data/RecordReaderBuilder.cs

```diff
@@ -16,8 +20,8 @@ namespace LinqToDB.Data
 
 	class RecordReaderBuilder
 	{
-		public static readonly ParameterExpression DataReaderParam  = Expression.Parameter(typeof(IDataReader),  "rd");
-		public        readonly ParameterExpression DataReaderLocal;
+		public static   ParameterExpression DataReaderParam  = Expression.Parameter(typeof(IDataReader),  "rd");
+		public readonly ParameterExpression DataReaderLocal;
```

Comment:
Changing a `public static readonly` field to be `public static` can introduce bugs in the future. Currently, the internal expectation is that no other classes (including in l2db classes) will change these fields; however, dropping the `readonly` allows other classes to change them.

I would recommend changing to the following:
```suggestion
		public  static  ParameterExpression DataReaderParam  => DataReaderLocal;
		private static ParameterExpression DataReaderLocal   =  Expression.Parameter(typeof(IDataReader), "rd");
```

And updating `ClearCache` to update the private field instead.

Reply:
Makes sense but changing a field to a property is a binary breaking change, correct?

Reply:
Technically yes, but only for code directly accessing it. Since this class (and the other one commented) are both `internal`, then we can safely assume that no consumer code accesses these fields. As such, it would not be an externally breaking change.

However, this is why all `public` values should be properties, even if the class is internal and the property is an auto-property that will get JIT-ted away - it means that implementation can change to a regular field if changes are necessary.